### PR TITLE
bin: add yargs and refactor

### PR DESCRIPTION
**Pull request by andrewhughes101**
- added yargs to simplify cli options
- added some tests and GitHub actions
- added a --total option which will limit how many dependents are fetched from the npm website
- added a --json option

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,16 +1,73 @@
...
+      forks: 100,
+      stars: 707,
+      watchers: 109,
+      url: 'github.com/firebase/firebase-functions'
+    }
+  }
+]
 ```
 
 Note you need a GtiHub personal token called GITHUB_TOKEN to use this tool. For
```
Comment (BethGriggs) [inline README.md]:
```suggestion
Note that to use this tool you need a GitHub personal token set as an environment variable named GITHUB_TOKEN. For
```

Typo and clarify how needs to be set.

File: package.json [hunk 2]
```diff
@@ -1,16 +1,28 @@
 {
-  "name": "dependents-testing",
+  "name": "dependents",
   "version": "0.1.0",
```
Comment (BethGriggs) [inline package.json]: Should we increment the version in this PR too?

File: package.json [hunk 3]
```diff
@@ -1,16 +1,28 @@
...
+  },
   "author": "Andrew Hughes",
   "license": "Apache-2.0",
-  "repository": "andrewhughes101/dependents-testing",
+  "repository": {
+    "type": "git",
+    "url": "git@github.com:pkgjs/dependents.git"
+  },
   "dependencies": {
     "@octokit/graphql": "^4.3.1",
```
Comment (BethGriggs) [inline package.json:17]: Is the GitHub GraphQL module still used? I couldn't spot it from my quick skim.

File: test/index.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
+require('dotenv').config()
+const assert = require('assert')
+const { suite, test } = require('mocha')
+const getDependents = require('../lib/getDependents')
+const pkg = require('../package.json')
+const config = require('./fixtures/config')
+
+assert(process.env.GITHUB_TOKEN, `
+Tests requires you to have a GitHub personal token called GITHUB_TOKEN
```
Comment (BethGriggs) [inline test/index.js]: requires -> require

**Comment by BethGriggs** [comment]
A few nits but other than those LGTM

File: package.json (hunk 3 above)
Comment (andrewhughes101) [reply package.json:17]: Yep used in `lib/github.js`

File: package.json (hunk 2 above)
Comment (andrewhughes101) [reply package.json]: Done

File: test/index.js (hunk 4 above)
Comment (andrewhughes101) [reply test/index.js]: Done

**Review by BethGriggs — APPROVED** [review]

**Pull request merged**